bentoml/_internal/frameworks/catboost.py

```diff
@@ -261,9 +258,21 @@ def _run(
             self: CatBoostRunnable,
             input_data: ext.NpNDArray | ext.CbPool | ext.PdDataFrame,
         ) -> ext.NpNDArray:
+            params_ = self.predict_params.copy()
```

Comment:
Can we avoid this copy by creating two `predict_params`, one for `predict` and one for `staged_predict`.